---
title: C6312 | Microsoft Docs
ms.custom: ''
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C6312
helpviewer_keywords:
- C6312
ms.assetid: 1fc8b9a1-e6ba-4799-84c3-31f289576cca
caps.latest.revision: 15
author: mikeblome
ms.author: mblome
manager: ghogen
ms.openlocfilehash: c2d627c4b8387e43b154ca6f2fb1d3ae334a70bd
ms.sourcegitcommit: af428c7ccd007e668ec0dd8697c88fc5d8bca1e2
ms.translationtype: MT
ms.contentlocale: tr-TR
ms.lasthandoff: 11/16/2018
ms.locfileid: "51816768"
---
# <a name="c6312"></a>C6312
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

Uyarı C6312: olası sonsuz döngü: try özel durum filtresi ifadesinde exceptıon_contınue_executıon sabiti kullanımı-dışında  
  
 Bu uyarı, sabit kullanımını gösterir. `EXCEPTION_CONTINUE_EXECUTION` (veya -1 olarak değerlendirilen başka bir sabit) yapılandırılmış özel durum işleyicisinin filtre ifadesinde. Sabit değer kullanımını `EXCEPTION_CONTINUE_EXECUTION` sonsuz bir döngüye neden olabilir. Örneğin, donanım tarafından bir özel durum oluştu, özel duruma neden olan yönergeyi yeniden başlatılır. Özel duruma neden olan adresi hala hatalı ise, başka bir özel durum ortaya çıkar ve aynı şekilde ele alınabilir. Bu, sonsuz bir döngüye neden olur.  
  
 Açık bir çağrı `RaiseException` doğrudan sonsuz bir döngüye neden olur, ancak bu kodun protected blokta devam edecektir. Bu beklenmeyen ve, sonsuz bir döngüye neden `RaiseException` geçersiz bir işaretçinin önlemek için kullanıldı.  
  
 Genellikle, `EXCEPTION_CONTINUE_EXECUTION` yalnızca özel duruma neden işaretçi ya da temel alınan bellek düzeltme şansı filtre ifadesinde çağrılan işlev tarafından döndürülen.  
  
## <a name="example"></a>Örnek  
 Aşağıdaki kod bu uyarıyı üretir:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *ptr)  
{  
  __try   
  {  
    // exception occurs if the caller passes null ptr  
    // code...     
   *ptr = '\0';  
  }   
  __except (EXCEPTION_CONTINUE_EXECUTION)  
    // When EXCEPTION_CONTINUE_EXECUTION is used, the handler  
    //  block of the structured exception handler is not executed.   
  {  
    puts("This block is never executed");  
  }  
}  
```  
  
 Bu uyarıyı düzeltmek için aşağıdaki kodu kullanın:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *ptr)  
{  
  __try   
  {  
    // exception occurs if the caller passes null ptr  
    // code...  
    *ptr = '\0';  
  }   
  __except (GetExceptionCode()== EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)   
  {  
    puts("Error Occurred");  
  }  
}  
```



